Psoc5lp

mcakmak opened the pull request:
This branch is created to port RIOT to cy8ckit-050 development kit which designed for Cypress PSoC 5LP IC.  
Following directories are added;
- board/cy8ckit-050
- cpu/psoc5lp
- dist/PSoCCreator

RIOT is ported to an empty PSoC Creator project.  To use this port, please open following file with PSoC Creator IDE (please download from http://www.cypress.com/psoccreator/);
 -> dist/PSoCCreator/RIOT.cywrk

This is a PSoC Project and developer only use RIOT port with this project. Default RIOT build environment cannot be used with this port; PSoC port does not include makefile's. 

PSoC IC's does not have some HW blocks (e.g. Timer, UART ) as default. Some simple HW blocks are created for RIOT. RIOT Project under dist/PSoCCreator already contains those HW designs. 

PSoC Creator IDE creates some file for device initialization. It is also creates files for created HW designs. (Auto-generated file will be created after build in PSoC Creator). board and cpu files use those files directly. This port did not try to re-implement all.

Please, open RIOT project with PSoC Creator, build and download to device. RIOT is going to work on PSoC 5LP. 

Note: Other development kit environments(LCD, Buttons, LEDs etc.) will be supported in the next revisions.

mcakmak committed:
PSoC Creator project which includes RIOT is created.

mcakmak committed:
Notes(for developers) removed from PSoC Creator project.

mcakmak committed:
cy8ckit-050 dev kit files are added.

mcakmak committed:
PSoC 5LP specific files are added.

mcakmak committed:
HW issue about clocking is fixed.

mcakmak committed:
HW Design issue about UART Clock is fixed.

kaspar030 commented on boards/cy8ckit-050/board.c:
```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright (C) 2014 Freie Universität Berlin
```
see below

kaspar030 commented on boards/cy8ckit-050/include/board.h:
```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright (C) 2014 INRIA
+ * Copyright (C) 2014 Freie UniversitÃ¤t Berlin
```
somehow here github doesn't show the umlaut correctly, but it does above. Wrong charset in one of the files? (should be utf-8 everywhere...)

mcakmak committed:
Uncorrect umlaut characters are fixed.

mcakmak committed:
- Boot files are added for PSoC 5LP port.
 - Makefile's are added for RIOT default build system.
 - PSoC Creator project is moved to dist/PSoCCretor/cy8ckit-050 folder to avoid other porting conflicts (e.g. psoc 4ble)

mcakmak committed:
Revert "Uncorrect umlaut characters are fixed."

This reverts commit 9cc3779ed4c7df0d10c7c6993551a6c6a4949e82.

mcakmak committed:
White space character issue is fixed.

mcakmak committed:
White spaces are removed with SED

mcakmak committed:
White spaces under PSoC Creator directory are removed

mcakmak committed:
White spaces under PSoC Creator directory are removed
